### DISCUSSION TITLE
How to dynamically set default header for all the request for tauri app

### DISCUSSION THREAD

**Discussion by crackleeeessyp** [tauri-apps/tauri#11807] [Q&A] [answered]
Seems currently we can setup default header in tauri.config.json file, but how can I set it dynamically in rust?

**Comment by FabianLars** [comment, marked as answer]
1) those headers are only set on requests to the tauri built-in protocols. if you're asking for general requests to a non-tauri url then this doesn't help
2) only support in tauri.conf.json, see https://github.com/tauri-apps/tauri/pull/11485/files. That said, you can modify the config in rust before the tauri::Builder line similar to https://github.com/tauri-apps/tauri-plugin-localhost#usage

**Reply by crackleeeessyp** [reply]
> you can modify the config in rust

OK, thanks! It is useful to change config in runtime.

I'm looking for a way to set headers for all network requests. I hope a future version will include this feature. In Electron, I can use Session to achieve this.

**Reply by FabianLars** [reply]
This won't be possible as long as we use the system webviews. They simply do not allow this (except Windows).

**Reply by crackleeeessyp** [reply]
Not sure how electron can intercept request, and I think android platform can also intercept request

**Reply by FabianLars** [reply]
Electron uses their own fork of chromium so they have full control over the webengine. We rely on the one included in the OS and have to live with what they allow us to do.

> and I think android platform can also intercept request

could be yeah, but then we're at 2 out of 5 platforms :/

**Reply by ahirner** [reply]
@crackleeeessyp You sure asked for response and not request headers?  The latter is supported by `wry` as https://docs.rs/wry/0.54.1/wry/struct.WebView.html#method.load_url_with_headers. It's not exposed by tauri.
